airflow/contrib/sensors/redis_pub_sub_sensor.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class RedisPubSubSensor(BaseSensorOperator):
+    """
+    Redis sensor for reading a message from pub sub channels
+    """
+    template_fields = ('channels',)
+    ui_color = '#f0eee4'
+
+    @apply_defaults
+    def __init__(self, channels, hook, *args, **kwargs):
```

Comment:
I think we should use `redis_conn_id` instand of `hook`. we already have `redis_hook` in `airflow.contrib.hook.redis_hook` we could use it.
```suggestion
    def __init__(self, channels, redis_conn_id, *args, **kwargs):
```
than we could create connection using 
```python
self.hook = RedisHook(redis_conn_id=self.redis_conn_id)
```

Reply:
Hi @zhongjiajie - I was trying the decouple the responsibility of creating the hook from using it by the sensor . This will allow injection of the hook and use mock hooks for unit testing.  
Let me know your thoughts.

Reply:
@mans2singh I think we should use conn str rather than hook as parameter. airflow provide function to create hook by conn str.

Reply:
@zhongjiajie - I have seen that pattern of passing a connection id and the sensor/operator creating the hook and using it.   I was trying to use dependency injection pattern to make it easy to inject a mock hook and test the sensor.  Is there any disadvantage of using this pattern ?

Reply:
@mans2singh disadvantage is we have to write code to create hook in DAG file. I think not high cohesion

If we pass conn_id str, user could just pass conn_id str to active sensors. For unittest I think we could do like other sensors.

Reply:
Changed to use redis_conn_id as recommended.